# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal; autoconf; autoheader; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/include
install -m 644 gmp.h $DESTDIR/include
install -d $DESTDIR/lib
install .libs/libgmp.so.10.* $DESTDIR/lib/libgmp.so.10
ln -sf libgmp.so.10 $DESTDIR/lib/libgmp.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 gmp.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/gmp-build.pc << EOF
Name: gmp
Version: 6.2.1
Description: GNU Multiple Precision arithmetic library
Requires: $REQUIRES

devel=\\
/include/gmp.h \\
/lib/libgmp.so \\
/lib/pkgconfig/gmp.pc

exec=\\
/lib/libgmp.so.10
EOF

exit
--- gmp-6.2.1.orig/configure.ac
+++ gmp-6.2.1/configure.ac
@@ -39,6 +39,7 @@
 AC_PREREQ(2.59)
 AC_INIT(GNU MP, GMP_VERSION, [gmp-bugs@gmplib.org, see https://gmplib.org/manual/Reporting-Bugs.html], gmp)
 AC_CONFIG_SRCDIR(gmp-impl.h)
+AC_CONFIG_MACRO_DIR([m4])
 m4_pattern_forbid([^[ \t]*GMP_])
 m4_pattern_allow(GMP_LDFLAGS)
 m4_pattern_allow(GMP_LIMB_BITS)
